### DISCUSSION TITLE
Commands fuzzy search: I don't really understand how it works

### DISCUSSION THREAD

**Discussion by aborruso** [saulpw/visidata#2643] [General]
Hi to all.

If I click on spacebar and then I type "sele arou," among the selected commands I don't have "select-around-n".

![image](https://github.com/user-attachments/assets/fb3e1408-4b84-46aa-8ea1-864c05f51516)

It seems very strange to me, because I entered so many characters "close" to it, and even in the fuzzy search the command "select-around-n" should be proposed to me among the first.
What didn't I understand about the way it search? Is there any way to make it work so that "select-around-n" appears higher in the list?

Thank you

**Comment by midichef** [comment]
Try removing the space: searching for `selearou` will find `select-around-n` as the top result.

It would be nice to make the command palette handle spaces as we'd expect. And I have noticed one or two more search problems in the command palette myself. But its `fuzzymatch` code seems too complicated for me to look into.

**Reply by midichef** [reply]
Okay I opened an issue for the command search:  #2645. If you find any other odd fuzzy match results, please add them to that issue. Thank you for reporting this.

**Reply by aborruso** [reply]
Thank you very much!!

**Reply by midichef** [reply]
This turned out to be easy to figure out. You can monitor the fix at #2646.

Your example of `sele arou` made it very easy to track down what was wrong. When I've been confused by similar behavior in the past, I just figured search was a bit unintuitive. But it was actually buggy. Thanks for speaking up with a good repro!
